src/backend/gporca/libgpos/include/gpos/common/CAutoRg.h

```diff
@@ -33,10 +33,9 @@ class CAutoRg : public CStackObject
 	// actual element to point to
 	T *m_object_array;
 
-	// hidden copy ctor
-	CAutoRg<T>(const CAutoRg &);
-
 public:
+	CAutoRg<T>(const CAutoRg &) = delete;
```

Comment:
The change fix this warning:
```sh
/home/gpadmin/gpdb_src/src/backend/gporca/libgpos/server/src/unittest/gpos/common/CAutoRgTest.cpp: In static member function 'static gpos::GPOS_RESULT gpos::CAutoRgTest::EresUnittest_Basics()':
/home/gpadmin/gpdb_src/src/backend/gporca/libgpos/server/src/unittest/gpos/common/CAutoRgTest.cpp:63:16: error: implicitly-declared 'gpos::CAutoRg<char>& gpos::CAutoRg<char>::operator=(const gpos::CAutoRg<char>&)' is deprecated [-Werror=deprecated-copy]
   63 |         asz3 = asz2;
      |                ^~~~
In file included from /home/gpadmin/gpdb_src/src/backend/gporca/libgpos/server/src/unittest/gpos/common/CAutoRgTest.cpp:15:
/home/gpadmin/gpdb_src/src/backend/gporca/libgpos/include/gpos/common/CAutoRg.h:37:9: note: because 'gpos::CAutoRg<char>' has user-provided 'gpos::CAutoRg<T>::CAutoRg(const gpos::CAutoRg<T>&) [with T = char]'
   37 |         CAutoRg<T>(const CAutoRg &);
      |         ^~~~~~~~~~
```
I took this change from GPDB 7.
See explanation in previous message.

Reply:
The same question as in previous message: can we delete the constructor instead of marking it?

Reply:
See my [previous](https://github.com/arenadata/gpdb/pull/1094/files/49872709156baa31a3bf3044b20608410231c301#r1833752436) message.